src/main/osd/osd.c

```diff
@@ -386,15 +377,34 @@ void osdInit(displayPort_t *osdDisplayPortToUse)
     }
 #endif
 
-    displayResync(osdDisplayPort);
-
     resumeRefreshAt = micros() + (4 * REFRESH_1S);
 #ifdef USE_OSD_PROFILES
     setOsdProfile(osdConfig()->osdProfileIndex);
 #endif
 
     osdElementsInit(backgroundLayerSupported);
     osdAnalyzeActiveElements();
+    displayCommitTransaction(osdDisplayPort);
+
+    osdIsReady = true;
+}
+
+void osdInit(displayPort_t *osdDisplayPortToUse)
+{
+    if (!osdDisplayPortToUse) {
+        return;
+    }
+
+    STATIC_ASSERT(OSD_POS_MAX == OSD_POS(31,31), OSD_POS_MAX_incorrect);
```

Comment:
I think `STATIC_ASSERT()` is actually static these days, so this can be moved outside of a function body.